spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BatchNormalization.scala

```diff
@@ -243,25 +296,39 @@ class BatchNormalization[T: ClassTag](
   }
 
   override def updateGradInput(input: Tensor[T], gradOutput: Tensor[T]): Tensor[T] = {
+    val gmKeyWithId = s"${this.gmKey}_${this.getId}"
+    val gxmKeyWithId = s"${this.gxmKey}_${this.getId}"
+    val needSync = getParallism() != None && getParallism().get > 1
     _gradOutput.set(gradOutput)
     makeBatch(_gradOutput)
     _gradOutput.addSingletonDimension(_gradOutput, 3)
     _gradOutput.addSingletonDimension(_gradOutput, 4)
     gxMean.zero()
     gMean.zero()
+    val nChannel = _gradOutput.size(2)
```

Comment:
should we require the input is NCHW?

Reply:
it's already separated in NCHW related methods, not impact on others